synapse/lib/aha.py

```diff
@@ -132,10 +132,18 @@ async def addAhaClone(self, host, port=27492, conf=None):
 
     async def getAhaUrls(self, user='root'):
         ahaurls = await self.cell.getAhaUrls(user=user)
-        if ahaurls is None:
-            return ()
         return ahaurls
 
+    @s_cell.adminapi()
+    async def callAhaPeerApi(self, iden, todo, timeout=None, skiprun=None):
+        async for item in self.cell.callAhaPeerApi(iden, todo, timeout=timeout, skiprun=skiprun):
```

Comment:
Since service users connecting to AHA service may not be root ( for example, provisioned with `--user foouser` ) we'll have to resolve the service iden to identify what services they are, and then and then check the `aha.service.get.<network>.<service>` values which are being connected to.

Reply:
This API requires admin because the AHA service connects to the target service on your behalf but as root. Allowing anybody with `aha.service.get` would introduce a priv escalation vuln.

Reply:
Then this functionality will not be available to deployments which were made with services provisioned like the following:

```
python -m synapse.tools.aha.provision.service 00.fooservice --user someuser
```

For example, the Cortex would then be connecting to AHA using the user `someuser` which does not have admin permissions to call the peer apis.

Reply:
Correct, because allowing that user to dispatch API calls as root on those other services should not be allowed by simply being able to see the service. The solution is to make `someuser` an admin on the AHA service.